sparse/src/KokkosSparse_CooMatrix.hpp

```diff
@@ -0,0 +1,129 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+/// \file KokkosSparse_CooMatrix.hpp
+/// \brief Local sparse matrix interface
+///
+/// This file provides KokkosSparse::CooMatrix.  This implements a
+/// local (no MPI) sparse matrix stored in coordinate ("Coo") format
+/// which is also known as ivj or triplet format.
+
+#ifndef KOKKOS_SPARSE_COOMATRIX_HPP_
+#define KOKKOS_SPARSE_COOMATRIX_HPP_
+
+#include "Kokkos_Core.hpp"
+#include "KokkosKernels_Error.hpp"
+#include <sstream>
+#include <stdexcept>
+
+namespace KokkosSparse {
+/// \class CooMatrix
+///
+/// \brief Coordinate format implementation of a sparse matrix.
+///
+/// \tparam RowView    The type of row index view.
+/// \tparam ColumnView The type of column index view.
+/// \tparam DataView   The type of data view.
+/// \tparam Device     The Kokkos Device type.
+/// \tparam MemoryTraits Traits describing how Kokkos manages and
+///   accesses data.  The default parameter suffices for most users.
+///
+/// "Coo" stands for "coordinate format".
+template <class RowView, class ColumnView, class DataView, class Device>
```

Comment:
Is there a reason to have the template params differ from CrsMatrix and CcsMatrix? If this took ``<Scalar, Ordinal, Device, Traits, SizeType>`` then size_type wouldn't have to be hardcoded, and ``row_type`` and ``column_type`` could both just be ``Ordinal``.